File: offer-storage-service/src/main/java/pl/allezon/offerstorage/persistence/OfferRepository.java
package pl.allezon.offerstorage.persistence;

import pl.allezon.offerstorage.domain.Offer;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OfferRepository extends JpaRepository<Offer, Long> {
}

File: offer-cache-service/src/main/scala/pl/allezon/offercache/repository/OfferRepository.scala
package pl.allezon.offercache.repository

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import pl.allezon.offercache.domain.Offer

@Repository
trait OfferRepository extends MongoRepository[Offer, String] {
  def findById(id: String): Offer
}
